triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/dao/DbTriggerDao.scala

```diff
@@ -38,6 +39,15 @@ abstract class DbTriggerDao protected (
   protected implicit def uuidPut: Put[UUID]
   protected implicit def uuidGet: Get[UUID]
 
+  implicit val readAsPut: Put[Set[Party]] =
+    implicitly[Put[String]].contramap { (readAs: Set[Party]) =>
+      readAs.map(Tag.unwrap).mkString(",")
+    }
+
+  implicit val readAsGet: Get[Set[Party]] = implicitly[Get[String]].map {
+    _.split(",").map(_.trim).toSet.filter(_.nonEmpty).map(Party.apply)
+  }
```

Comment:
```suggestion
  implicit val readAsGet: Get[Set[Party]] = {
    type F[A] = Get[Set[A]]
    Party.subst[F, String](implicitly[Get[String]].map {
      _.split(",").map(_.trim).toSet.filter(_.nonEmpty)
    }
  }
```

Reply:
Is there a specific reason why this code is better than the other?

Reply:
This does subst once for the whole program, the original code does subst once for each party that is read every time it is read.

Reply:
Nice, thanks for the info!